timemachine/maps/estimators.py

```diff
@@ -0,0 +1,56 @@
+"""Utilities for computing mapped works or mapped u_kn matrix, incorporating invertible map_fxns"""
+
+import numpy as np
+from pymbar.utils import kln_to_kn  # TODO: should this be "vendored"?
+
+__all__ = ["compute_mapped_reduced_work", "compute_mapped_u_kn"]
+
+
+def compute_mapped_reduced_work(src_samples, src_reduced_energy_fxn, dst_reduced_energy_fxn, map_fxn):
+    """Compute minus log importance weights, including a change of variables defined by map_fxn
+
+    References
+    ----------
+    [Jarzynski, 2002] Targeted free energy perturbation
+        https://journals.aps.org/pre/abstract/10.1103/PhysRevE.65.046122
+    """
+
+    # compare with version without map_fxn
+    # return dst_reduced_energy_fxn(src_samples) - src_reduced_energy_fxn(src_samples)
+
+    mapped_samples, logdetjacs = map_fxn(src_samples)
+    return dst_reduced_energy_fxn(mapped_samples) - src_reduced_energy_fxn(src_samples) - logdetjacs
+
+
+def compute_mapped_u_kn(sample_lists, reduced_energy_fxns, map_fxns):
+    """Evaluate every sample from state i in every state j,
+    incorporating user-defined change of variables supplied in map_fxns[i, j]
```

Comment:
Possible documentation / runtime check to be added: This function assumes (without verifying) that:
(1) `map_fxns[i, j]` is the inverse of `map_fxns[j, i]`,
(2) the domain of `map_fxns[i]` is the support of state `i`,
(3) the codomain of `map_fxns[i, j]` is the support of state `j`.